### PR TITLE
feat(route): add voronoiapp

### PR DESCRIPTION
**Pull request by cesaryuan**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/voronoiapp/latest
/voronoiapp/latest/Automotive
/voronoiapp/latest/Business
/voronoiapp/latest/Climate
/voronoiapp/latest/Demographics
/voronoiapp/latest/Economy
/voronoiapp/latest/Energy
/voronoiapp/latest/Entertainment
/voronoiapp/latest/Geopolitics
/voronoiapp/latest/Healthcare
/voronoiapp/latest/Innovation
/voronoiapp/latest/Maps
/voronoiapp/latest/Markets
/voronoiapp/latest/Money
/voronoiapp/latest/Natural Resources
/voronoiapp/latest/Politics
/voronoiapp/latest/Public Opinion
/voronoiapp/latest/Real Estate
/voronoiapp/latest/Sports
/voronoiapp/latest/Technology
/voronoiapp/latest/Wealth
/voronoiapp/latest/Travel
/voronoiapp/latest/Nature
/voronoiapp/latest/Space
/voronoiapp/latest/Diagram
/voronoiapp/latest/Other
/voronoiapp/popular
/voronoiapp/popular/most-popular
/voronoiapp/popular/most-popular/WEEK
/voronoiapp/popular/most-popular/WEEK/Automotive
/voronoiapp/popular/most-popular/WEEK/Business
/voronoiapp/popular/most-popular/WEEK/Climate
/voronoiapp/popular/most-popular/WEEK/Demographics
/voronoiapp/popular/most-popular/WEEK/Economy
/voronoiapp/popular/most-popular/WEEK/Energy
/voronoiapp/popular/most-popular/WEEK/Entertainment
/voronoiapp/popular/most-popular/WEEK/Geopolitics
/voronoiapp/popular/most-popular/WEEK/Healthcare
/voronoiapp/popular/most-popular/WEEK/Innovation
/voronoiapp/popular/most-popular/WEEK/Maps
/voronoiapp/popular/most-popular/WEEK/Markets
/voronoiapp/popular/most-popular/WEEK/Money
/voronoiapp/popular/most-popular/WEEK/Natural Resources
/voronoiapp/popular/most-popular/WEEK/Politics
/voronoiapp/popular/most-popular/WEEK/Public Opinion
/voronoiapp/popular/most-popular/WEEK/Real Estate
/voronoiapp/popular/most-popular/WEEK/Sports
/voronoiapp/popular/most-popular/WEEK/Technology
/voronoiapp/popular/most-popular/WEEK/Wealth
/voronoiapp/popular/most-popular/WEEK/Travel
/voronoiapp/popular/most-popular/WEEK/Nature
/voronoiapp/popular/most-popular/WEEK/Space
/voronoiapp/popular/most-popular/WEEK/Diagram
/voronoiapp/popular/most-popular/WEEK/Other
/voronoiapp/popular/most-popular/MONTH
/voronoiapp/popular/most-popular/MONTH/Automotive
/voronoiapp/popular/most-popular/MONTH/Business
/voronoiapp/popular/most-popular/MONTH/Climate
/voronoiapp/popular/most-popular/MONTH/Demographics
/voronoiapp/popular/most-popular/MONTH/Economy
/voronoiapp/popular/most-popular/MONTH/Energy
/voronoiapp/popular/most-popular/MONTH/Entertainment
/voronoiapp/popular/most-popular/MONTH/Geopolitics
/voronoiapp/popular/most-popular/MONTH/Healthcare
/voronoiapp/popular/most-popular/MONTH/Innovation
/voronoiapp/popular/most-popular/MONTH/Maps
/voronoiapp/popular/most-popular/MONTH/Markets
/voronoiapp/popular/most-popular/MONTH/Money
/voronoiapp/popular/most-popular/MONTH/Natural Resources
/voronoiapp/popular/most-popular/MONTH/Politics
/voronoiapp/popular/most-popular/MONTH/Public Opinion
/voronoiapp/popular/most-popular/MONTH/Real Estate
/voronoiapp/popular/most-popular/MONTH/Sports
/voronoiapp/popular/most-popular/MONTH/Technology
/voronoiapp/popular/most-popular/MONTH/Wealth
/voronoiapp/popular/most-popular/MONTH/Travel
/voronoiapp/popular/most-popular/MONTH/Nature
/voronoiapp/popular/most-popular/MONTH/Space
/voronoiapp/popular/most-popular/MONTH/Diagram
/voronoiapp/popular/most-popular/MONTH/Other
/voronoiapp/popular/most-popular/YEAR
/voronoiapp/popular/most-popular/YEAR/Automotive
/voronoiapp/popular/most-popular/YEAR/Business
/voronoiapp/popular/most-popular/YEAR/Climate
/voronoiapp/popular/most-popular/YEAR/Demographics
/voronoiapp/popular/most-popular/YEAR/Economy
/voronoiapp/popular/most-popular/YEAR/Energy
/voronoiapp/popular/most-popular/YEAR/Entertainment
/voronoiapp/popular/most-popular/YEAR/Geopolitics
/voronoiapp/popular/most-popular/YEAR/Healthcare
/voronoiapp/popular/most-popular/YEAR/Innovation
/voronoiapp/popular/most-popular/YEAR/Maps
/voronoiapp/popular/most-popular/YEAR/Markets
/voronoiapp/popular/most-popular/YEAR/Money
/voronoiapp/popular/most-popular/YEAR/Natural Resources
/voronoiapp/popular/most-popular/YEAR/Politics
/voronoiapp/popular/most-popular/YEAR/Public Opinion
/voronoiapp/popular/most-popular/YEAR/Real Estate
/voronoiapp/popular/most-popular/YEAR/Sports
/voronoiapp/popular/most-popular/YEAR/Technology
/voronoiapp/popular/most-popular/YEAR/Wealth
/voronoiapp/popular/most-popular/YEAR/Travel
/voronoiapp/popular/most-popular/YEAR/Nature
/voronoiapp/popular/most-popular/YEAR/Space
/voronoiapp/popular/most-popular/YEAR/Diagram
/voronoiapp/popular/most-popular/YEAR/Other
/voronoiapp/popular/most-popular/ALL
/voronoiapp/popular/most-popular/ALL/Automotive
/voronoiapp/popular/most-popular/ALL/Business
/voronoiapp/popular/most-popular/ALL/Climate
/voronoiapp/popular/most-popular/ALL/Demographics
/voronoiapp/popular/most-popular/ALL/Economy
/voronoiapp/popular/most-popular/ALL/Energy
/voronoiapp/popular/most-popular/ALL/Entertainment
/voronoiapp/popular/most-popular/ALL/Geopolitics
/voronoiapp/popular/most-popular/ALL/Healthcare
/voronoiapp/popular/most-popular/ALL/Innovation
/voronoiapp/popular/most-popular/ALL/Maps
/voronoiapp/popular/most-popular/ALL/Markets
/voronoiapp/popular/most-popular/ALL/Money
/voronoiapp/popular/most-popular/ALL/Natural Resources
/voronoiapp/popular/most-popular/ALL/Politics
/voronoiapp/popular/most-popular/ALL/Public Opinion
/voronoiapp/popular/most-popular/ALL/Real Estate
/voronoiapp/popular/most-popular/ALL/Sports
/voronoiapp/popular/most-popular/ALL/Technology
/voronoiapp/popular/most-popular/ALL/Wealth
/voronoiapp/popular/most-popular/ALL/Travel
/voronoiapp/popular/most-popular/ALL/Nature
/voronoiapp/popular/most-popular/ALL/Space
/voronoiapp/popular/most-popular/ALL/Diagram
/voronoiapp/popular/most-popular/ALL/Other
/voronoiapp/popular/most-discussed
/voronoiapp/popular/most-discussed/WEEK
/voronoiapp/popular/most-discussed/WEEK/Automotive
/voronoiapp/popular/most-discussed/WEEK/Business
/voronoiapp/popular/most-discussed/WEEK/Climate
/voronoiapp/popular/most-discussed/WEEK/Demographics
/voronoiapp/popular/most-discussed/WEEK/Economy
/voronoiapp/popular/most-discussed/WEEK/Energy
/voronoiapp/popular/most-discussed/WEEK/Entertainment
/voronoiapp/popular/most-discussed/WEEK/Geopolitics
/voronoiapp/popular/most-discussed/WEEK/Healthcare
/voronoiapp/popular/most-discussed/WEEK/Innovation
/voronoiapp/popular/most-discussed/WEEK/Maps
/voronoiapp/popular/most-discussed/WEEK/Markets
/voronoiapp/popular/most-discussed/WEEK/Money
/voronoiapp/popular/most-discussed/WEEK/Natural Resources
/voronoiapp/popular/most-discussed/WEEK/Politics
/voronoiapp/popular/most-discussed/WEEK/Public Opinion
/voronoiapp/popular/most-discussed/WEEK/Real Estate
/voronoiapp/popular/most-discussed/WEEK/Sports
/voronoiapp/popular/most-discussed/WEEK/Technology
/voronoiapp/popular/most-discussed/WEEK/Wealth
/voronoiapp/popular/most-discussed/WEEK/Travel
/voronoiapp/popular/most-discussed/WEEK/Nature
/voronoiapp/popular/most-discussed/WEEK/Space
/voronoiapp/popular/most-discussed/WEEK/Diagram
/voronoiapp/popular/most-discussed/WEEK/Other
/voronoiapp/popular/most-discussed/MONTH
/voronoiapp/popular/most-discussed/MONTH/Automotive
/voronoiapp/popular/most-discussed/MONTH/Business
/voronoiapp/popular/most-discussed/MONTH/Climate
/voronoiapp/popular/most-discussed/MONTH/Demographics
/voronoiapp/popular/most-discussed/MONTH/Economy
/voronoiapp/popular/most-discussed/MONTH/Energy
/voronoiapp/popular/most-discussed/MONTH/Entertainment
/voronoiapp/popular/most-discussed/MONTH/Geopolitics
/voronoiapp/popular/most-discussed/MONTH/Healthcare
/voronoiapp/popular/most-discussed/MONTH/Innovation
/voronoiapp/popular/most-discussed/MONTH/Maps
/voronoiapp/popular/most-discussed/MONTH/Markets
/voronoiapp/popular/most-discussed/MONTH/Money
/voronoiapp/popular/most-discussed/MONTH/Natural Resources
/voronoiapp/popular/most-discussed/MONTH/Politics
/voronoiapp/popular/most-discussed/MONTH/Public Opinion
/voronoiapp/popular/most-discussed/MONTH/Real Estate
/voronoiapp/popular/most-discussed/MONTH/Sports
/voronoiapp/popular/most-discussed/MONTH/Technology
/voronoiapp/popular/most-discussed/MONTH/Wealth
/voronoiapp/popular/most-discussed/MONTH/Travel
/voronoiapp/popular/most-discussed/MONTH/Nature
/voronoiapp/popular/most-discussed/MONTH/Space
/voronoiapp/popular/most-discussed/MONTH/Diagram
/voronoiapp/popular/most-discussed/MONTH/Other
/voronoiapp/popular/most-discussed/YEAR
/voronoiapp/popular/most-discussed/YEAR/Automotive
/voronoiapp/popular/most-discussed/YEAR/Business
/voronoiapp/popular/most-discussed/YEAR/Climate
/voronoiapp/popular/most-discussed/YEAR/Demographics
/voronoiapp/popular/most-discussed/YEAR/Economy
/voronoiapp/popular/most-discussed/YEAR/Energy
/voronoiapp/popular/most-discussed/YEAR/Entertainment
/voronoiapp/popular/most-discussed/YEAR/Geopolitics
/voronoiapp/popular/most-discussed/YEAR/Healthcare
/voronoiapp/popular/most-discussed/YEAR/Innovation
/voronoiapp/popular/most-discussed/YEAR/Maps
/voronoiapp/popular/most-discussed/YEAR/Markets
/voronoiapp/popular/most-discussed/YEAR/Money
/voronoiapp/popular/most-discussed/YEAR/Natural Resources
/voronoiapp/popular/most-discussed/YEAR/Politics
/voronoiapp/popular/most-discussed/YEAR/Public Opinion
/voronoiapp/popular/most-discussed/YEAR/Real Estate
/voronoiapp/popular/most-discussed/YEAR/Sports
/voronoiapp/popular/most-discussed/YEAR/Technology
/voronoiapp/popular/most-discussed/YEAR/Wealth
/voronoiapp/popular/most-discussed/YEAR/Travel
/voronoiapp/popular/most-discussed/YEAR/Nature
/voronoiapp/popular/most-discussed/YEAR/Space
/voronoiapp/popular/most-discussed/YEAR/Diagram
/voronoiapp/popular/most-discussed/YEAR/Other
/voronoiapp/popular/most-discussed/ALL
/voronoiapp/popular/most-discussed/ALL/Automotive
/voronoiapp/popular/most-discussed/ALL/Business
/voronoiapp/popular/most-discussed/ALL/Climate
/voronoiapp/popular/most-discussed/ALL/Demographics
/voronoiapp/popular/most-discussed/ALL/Economy
/voronoiapp/popular/most-discussed/ALL/Energy
/voronoiapp/popular/most-discussed/ALL/Entertainment
/voronoiapp/popular/most-discussed/ALL/Geopolitics
/voronoiapp/popular/most-discussed/ALL/Healthcare
/voronoiapp/popular/most-discussed/ALL/Innovation
/voronoiapp/popular/most-discussed/ALL/Maps
/voronoiapp/popular/most-discussed/ALL/Markets
/voronoiapp/popular/most-discussed/ALL/Money
/voronoiapp/popular/most-discussed/ALL/Natural Resources
/voronoiapp/popular/most-discussed/ALL/Politics
/voronoiapp/popular/most-discussed/ALL/Public Opinion
/voronoiapp/popular/most-discussed/ALL/Real Estate
/voronoiapp/popular/most-discussed/ALL/Sports
/voronoiapp/popular/most-discussed/ALL/Technology
/voronoiapp/popular/most-discussed/ALL/Wealth
/voronoiapp/popular/most-discussed/ALL/Travel
/voronoiapp/popular/most-discussed/ALL/Nature
/voronoiapp/popular/most-discussed/ALL/Space
/voronoiapp/popular/most-discussed/ALL/Diagram
/voronoiapp/popular/most-discussed/ALL/Other
/voronoiapp/popular/most-viewed
/voronoiapp/popular/most-viewed/WEEK
/voronoiapp/popular/most-viewed/WEEK/Automotive
/voronoiapp/popular/most-viewed/WEEK/Business
/voronoiapp/popular/most-viewed/WEEK/Climate
/voronoiapp/popular/most-viewed/WEEK/Demographics
/voronoiapp/popular/most-viewed/WEEK/Economy
/voronoiapp/popular/most-viewed/WEEK/Energy
/voronoiapp/popular/most-viewed/WEEK/Entertainment
/voronoiapp/popular/most-viewed/WEEK/Geopolitics
/voronoiapp/popular/most-viewed/WEEK/Healthcare
/voronoiapp/popular/most-viewed/WEEK/Innovation
/voronoiapp/popular/most-viewed/WEEK/Maps
/voronoiapp/popular/most-viewed/WEEK/Markets
/voronoiapp/popular/most-viewed/WEEK/Money
/voronoiapp/popular/most-viewed/WEEK/Natural Resources
/voronoiapp/popular/most-viewed/WEEK/Politics
/voronoiapp/popular/most-viewed/WEEK/Public Opinion
/voronoiapp/popular/most-viewed/WEEK/Real Estate
/voronoiapp/popular/most-viewed/WEEK/Sports
/voronoiapp/popular/most-viewed/WEEK/Technology
/voronoiapp/popular/most-viewed/WEEK/Wealth
/voronoiapp/popular/most-viewed/WEEK/Travel
/voronoiapp/popular/most-viewed/WEEK/Nature
/voronoiapp/popular/most-viewed/WEEK/Space
/voronoiapp/popular/most-viewed/WEEK/Diagram
/voronoiapp/popular/most-viewed/WEEK/Other
/voronoiapp/popular/most-viewed/MONTH
/voronoiapp/popular/most-viewed/MONTH/Automotive
/voronoiapp/popular/most-viewed/MONTH/Business
/voronoiapp/popular/most-viewed/MONTH/Climate
/voronoiapp/popular/most-viewed/MONTH/Demographics
/voronoiapp/popular/most-viewed/MONTH/Economy
/voronoiapp/popular/most-viewed/MONTH/Energy
/voronoiapp/popular/most-viewed/MONTH/Entertainment
/voronoiapp/popular/most-viewed/MONTH/Geopolitics
/voronoiapp/popular/most-viewed/MONTH/Healthcare
/voronoiapp/popular/most-viewed/MONTH/Innovation
/voronoiapp/popular/most-viewed/MONTH/Maps
/voronoiapp/popular/most-viewed/MONTH/Markets
/voronoiapp/popular/most-viewed/MONTH/Money
/voronoiapp/popular/most-viewed/MONTH/Natural Resources
/voronoiapp/popular/most-viewed/MONTH/Politics
/voronoiapp/popular/most-viewed/MONTH/Public Opinion
/voronoiapp/popular/most-viewed/MONTH/Real Estate
/voronoiapp/popular/most-viewed/MONTH/Sports
/voronoiapp/popular/most-viewed/MONTH/Technology
/voronoiapp/popular/most-viewed/MONTH/Wealth
/voronoiapp/popular/most-viewed/MONTH/Travel
/voronoiapp/popular/most-viewed/MONTH/Nature
/voronoiapp/popular/most-viewed/MONTH/Space
/voronoiapp/popular/most-viewed/MONTH/Diagram
/voronoiapp/popular/most-viewed/MONTH/Other
/voronoiapp/popular/most-viewed/YEAR
/voronoiapp/popular/most-viewed/YEAR/Automotive
/voronoiapp/popular/most-viewed/YEAR/Business
/voronoiapp/popular/most-viewed/YEAR/Climate
/voronoiapp/popular/most-viewed/YEAR/Demographics
/voronoiapp/popular/most-viewed/YEAR/Economy
/voronoiapp/popular/most-viewed/YEAR/Energy
/voronoiapp/popular/most-viewed/YEAR/Entertainment
/voronoiapp/popular/most-viewed/YEAR/Geopolitics
/voronoiapp/popular/most-viewed/YEAR/Healthcare
/voronoiapp/popular/most-viewed/YEAR/Innovation
/voronoiapp/popular/most-viewed/YEAR/Maps
/voronoiapp/popular/most-viewed/YEAR/Markets
/voronoiapp/popular/most-viewed/YEAR/Money
/voronoiapp/popular/most-viewed/YEAR/Natural Resources
/voronoiapp/popular/most-viewed/YEAR/Politics
/voronoiapp/popular/most-viewed/YEAR/Public Opinion
/voronoiapp/popular/most-viewed/YEAR/Real Estate
/voronoiapp/popular/most-viewed/YEAR/Sports
/voronoiapp/popular/most-viewed/YEAR/Technology
/voronoiapp/popular/most-viewed/YEAR/Wealth
/voronoiapp/popular/most-viewed/YEAR/Travel
/voronoiapp/popular/most-viewed/YEAR/Nature
/voronoiapp/popular/most-viewed/YEAR/Space
/voronoiapp/popular/most-viewed/YEAR/Diagram
/voronoiapp/popular/most-viewed/YEAR/Other
/voronoiapp/popular/most-viewed/ALL
/voronoiapp/popular/most-viewed/ALL/Automotive
/voronoiapp/popular/most-viewed/ALL/Business
/voronoiapp/popular/most-viewed/ALL/Climate
/voronoiapp/popular/most-viewed/ALL/Demographics
/voronoiapp/popular/most-viewed/ALL/Economy
/voronoiapp/popular/most-viewed/ALL/Energy
/voronoiapp/popular/most-viewed/ALL/Entertainment
/voronoiapp/popular/most-viewed/ALL/Geopolitics
/voronoiapp/popular/most-viewed/ALL/Healthcare
/voronoiapp/popular/most-viewed/ALL/Innovation
/voronoiapp/popular/most-viewed/ALL/Maps
/voronoiapp/popular/most-viewed/ALL/Markets
/voronoiapp/popular/most-viewed/ALL/Money
/voronoiapp/popular/most-viewed/ALL/Natural Resources
/voronoiapp/popular/most-viewed/ALL/Politics
/voronoiapp/popular/most-viewed/ALL/Public Opinion
/voronoiapp/popular/most-viewed/ALL/Real Estate
/voronoiapp/popular/most-viewed/ALL/Sports
/voronoiapp/popular/most-viewed/ALL/Technology
/voronoiapp/popular/most-viewed/ALL/Wealth
/voronoiapp/popular/most-viewed/ALL/Travel
/voronoiapp/popular/most-viewed/ALL/Nature
/voronoiapp/popular/most-viewed/ALL/Space
/voronoiapp/popular/most-viewed/ALL/Diagram
/voronoiapp/popular/most-viewed/ALL/Other
/voronoiapp/home
/voronoiapp/home/Automotive
/voronoiapp/home/Business
/voronoiapp/home/Climate
/voronoiapp/home/Demographics
/voronoiapp/home/Economy
/voronoiapp/home/Energy
/voronoiapp/home/Entertainment
/voronoiapp/home/Geopolitics
/voronoiapp/home/Healthcare
/voronoiapp/home/Innovation
/voronoiapp/home/Maps
/voronoiapp/home/Markets
/voronoiapp/home/Money
/voronoiapp/home/Natural Resources
/voronoiapp/home/Politics
/voronoiapp/home/Public Opinion
/voronoiapp/home/Real Estate
/voronoiapp/home/Sports
/voronoiapp/home/Technology
/voronoiapp/home/Wealth
/voronoiapp/home/Travel
/voronoiapp/home/Nature
/voronoiapp/home/Space
/voronoiapp/home/Diagram
/voronoiapp/home/Other
/voronoiapp/editors-pick
/voronoiapp/editors-pick/Automotive
/voronoiapp/editors-pick/Business
/voronoiapp/editors-pick/Climate
/voronoiapp/editors-pick/Demographics
/voronoiapp/editors-pick/Economy
/voronoiapp/editors-pick/Energy
/voronoiapp/editors-pick/Entertainment
/voronoiapp/editors-pick/Geopolitics
/voronoiapp/editors-pick/Healthcare
/voronoiapp/editors-pick/Innovation
/voronoiapp/editors-pick/Maps
/voronoiapp/editors-pick/Markets
/voronoiapp/editors-pick/Money
/voronoiapp/editors-pick/Natural Resources
/voronoiapp/editors-pick/Politics
/voronoiapp/editors-pick/Public Opinion
/voronoiapp/editors-pick/Real Estate
/voronoiapp/editors-pick/Sports
/voronoiapp/editors-pick/Technology
/voronoiapp/editors-pick/Wealth
/voronoiapp/editors-pick/Travel
/voronoiapp/editors-pick/Nature
/voronoiapp/editors-pick/Space
/voronoiapp/editors-pick/Diagram
/voronoiapp/editors-pick/Other
/voronoiapp/search/china
/voronoiapp/author/visualcapitalist
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by cesaryuan** [comment]
@TonyRL Hi, sorry to bother you. This pr is ready for review. ❤️

**Comment by cesaryuan** [comment]
up

File: lib/routes/voronoiapp/search.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    name: 'Search Keyword Posts',
+    path: '/search/:keyword',
+    radar: [
+        {
+            source: ['www.voronoiapp.com/explore'],
+            // 新版本中貌似不再支持 function 形式的 target
+            target: (_, url) => {
+                const parsedURL = new URL(url);
+                const keyword = parsedURL.searchParams.get('search');
+                return `/search/${keyword}`;
```
Comment (TonyRL) [inline lib/routes/voronoiapp/search.ts:15]:
```suggestion
                return `/voronoiapp/search/${keyword}`;
```

File: lib/routes/voronoiapp/common.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    }
+    for (const key in finalSearchParams) {
+        if (finalSearchParams[key] !== undefined && finalSearchParams[key] !== null) {
+            url.searchParams.set(key, finalSearchParams[key]);
+        }
+    }
+    const data = await ofetch<Post[]>(url.toString());
+    const items: DataItem[] = data.map((post) => ({
+        title: post.headline,
+        link: `https://www.voronoiapp.com/${post.category.toLowerCase()}/${post.link}`,
```
[the remark this answers is not in the document]
Comment (cesaryuan) [reply lib/routes/voronoiapp/common.ts:58]: > In [Which Nation is the Most Happy?](https://www.voronoiapp.com/public-opinion/Which-Nation-is-the-Most-Happy--2522) from `/voronoiapp/author/visualcapitalist`, `post.category` is joined by `-` though a space still work.

Thank you for your amazing attention to detail!

**Pull request merged**